--- masqmail-0.3.4/configure.ac~	2012-01-18 09:31:45.000000000 +0100
+++ masqmail-0.3.4/configure.ac	2018-09-26 13:27:12.085535818 +0200
@@ -37,11 +37,9 @@ if test "$resolver_enabled" = yes; then
 dnl checks necessary for libc5:
 dnl if there is res_search in libc, it is probably libc5
 dnl if not, it is probably libc6 and we need libresolv
-	AC_CHECK_LIB(c, res_search, need_resolv=no, need_resolv=yes)
+	AC_CHECK_LIB(c, res_search, need_resolv=no, [AC_CHECK_LIB(c, __res_search, need_resolv=no, need_resolv=yes)])
 	if test "$need_resolv" = yes; then
-		AC_CHECK_LIB(resolv, res_search,
-			has_resolv=yes; RESOLV_LIBS="-lresolv",
-			has_resolv=no)
+		AC_CHECK_LIB(resolv, res_search,has_resolv=yes, [AC_CHECK_LIB(resolv, __res_search, has_resolv=yes; RESOLV_LIBS="-lresolv", [AC_MSG_ERROR("no libresolv")])])
 		if test "$has_resolv" = no; then
 			saved_LIBS="$LIBS"
 			LIBS="$LIBS -lresolv"
